packages/react-native/Libraries/Image/Image.android.js

```diff
@@ -40,19 +40,27 @@ function generateRequestId() {
  */
 function getSize(
   url: string,
-  success: (width: number, height: number) => void,
+  success?: (width: number, height: number) => void,
   failure?: (error: mixed) => void,
-): void {
-  NativeImageLoaderAndroid.getSize(url)
-    .then(function (sizes) {
-      success(sizes.width, sizes.height);
-    })
-    .catch(
-      failure ||
-        function () {
-          console.warn('Failed to get size for image: ' + url);
-        },
-    );
+): void | Promise<{width: number, height: number}> {
+  const isCallback = typeof success === 'function';
+  return isCallback
+    ? undefined
+    : NativeImageLoaderAndroid.getSize(url)
```

Comment:
What is this `undefined` return for? It seems like we wouldn't enter the `NativeImageLoaderAndroid.getSize` path anymore.

Reply:
right, my bad)